sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosContainer.java

```diff
@@ -175,8 +193,8 @@ public CosmosItemResponse upsertItem(Object item) throws CosmosClientException {
      * @return the cosmos sync item response
      * @throws CosmosClientException the cosmos client exception
      */
-    public CosmosItemResponse upsertItem(Object item, CosmosItemRequestOptions options) throws CosmosClientException {
-        return this.mapItemResponseAndBlock(this.containerWrapper.createItem(item, options));
+    public <T> CosmosItemResponse<T> upsertItem(Object item, CosmosItemRequestOptions options) throws CosmosClientException {
+        return (CosmosItemResponse<T>) this.mapItemResponseAndBlock(this.asyncContainer.upsertItem(item, options));
```

Comment:
I see `mapItemResponseAndBlock()` returns `CosmosItemResponse<T>`. 
Do we need this extra casting here ?

Reply:
Can live without it too, but left it for clarity. Will check and remove as required.